coral/src/app/features/components/table-filters/OperationTypeFilter.tsx

```diff
@@ -0,0 +1,58 @@
+import { NativeSelect } from "@aivenio/aquarium";
+import { useSearchParams } from "react-router-dom";
+import {
+  requestOperationTypeNameMap,
+  operationTypeList,
+} from "src/app/features/approvals/utils/request-operation-type-helper";
+import { RequestOperationType } from "src/domain/requests/requests-types";
+import { ResolveIntersectionTypes } from "types/utils";
+
+type RequestOperationTypeOptions = ResolveIntersectionTypes<
+  RequestOperationType | "ALL"
+>;
+
+function OperationTypeFilter() {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const operationType =
+    (searchParams.get("operationType") as RequestOperationTypeOptions | null) ??
+    "ALL";
+
+  const handleChangeOperationType = (
+    nextOperationType: RequestOperationTypeOptions
+  ) => {
+    if (nextOperationType === "ALL") {
+      searchParams.delete("operationType");
+      searchParams.set("page", "1");
+      setSearchParams(searchParams);
+    } else {
+      searchParams.set("operationType", nextOperationType);
+      searchParams.set("page", "1");
+      setSearchParams(searchParams);
+    }
+  };
+
+  return (
+    <NativeSelect
+      labelText={"Filter by operation type"}
+      key={"filter-operationType"}
+      defaultValue={operationType}
+      onChange={(e) => {
+        const operationType = e.target.value as RequestOperationTypeOptions;
+        return handleChangeOperationType(operationType);
```

Comment:
`handleChangeOperationType(e.target.value as RequestOperationTypeOptions))`? 

Also, i would even move the inline stuff into the handler function to make it even more clean:
`onChange={handleChangeOperationType}`

Reply:
Ah yes, good idea :o

[8a512a7](https://github.com/aiven/klaw/pull/888/commits/8a512a7c6831afd64ba004e0dcbbe0e0a687bf24)